docs/guide/controller-programming.md

```diff
@@ -113,35 +296,119 @@ Hence accessing the array beyond index 2 is illegal and may crash the controller
-x = pos[0];
-y = pos[1];
-z = pos[2];
+x = values[0];
+y = values[1];
+z = values[2];
 
 // OK, another way to copy the values
-double a[3] = { pos[0], pos[1], pos[2] };
+double a[3] = { values[0], values[1], values[2] };
 
 // OK, yet another way to copy these values
 double b[3];
-memcpy(b, pos, sizeof(b));
+memcpy(b, values, sizeof(b));
 ```
+%tab-end
+
+%tab "C++"
+```cpp
+const double *values = gps.getValues();
+
+// OK, to read the values they should never be explicitly deleted by the controller code
+std::cout << "MY_ROBOT is at position: " << values[0] << ' ' << values[1] << ' ' << values[2] << std::endl;
+
+// OK, to copy the values
+double x, y, z;
+x = values[0];
+y = values[1];
+z = values[2];
+
+// OK, another way to copy the values
+double a[3] = { values[0], values[1], values[2] };
+
+// OK, yet another way to copy these values (C++ 11)
+std::array<double, 3> b = { values[0], values[1], values[2] };
+```
+%tab-end
+
+%tab "Python"
+```python
+values = gps.getValues()
 
-And here are incorrect examples:
+# OK, to read the values they should never be explicitly deleted by the controller code
+print("MY_ROBOT is at position: %g %g %g" % (values[0], values[1], values[2]))
+
+# OK, to copy the values
+x = values[0]
+y = values[1]
+z = values[2]
+
+# OK, another way to copy the values
+a = []
+a = values
+
+# There is no really need to copy differently
+```
+%tab-end
+
+%tab "Java"
+```java
+final const values = gps.getValues();
+
+// OK, to read the values they should never be explicitly deleted by the controller code
+System.out.format("MY_ROBOT is at position: %g %g %g\n", values[0], values[1], values[2]);
+
+// OK, to copy the values
+double x, y, z;
+x = values[0];
+y = values[1];
+z = values[2];
+
+// OK, another way to copy the values
+double []a = values.clone();
+
+// OK, yet another way to copy these values
+double []b;
+System.arraycopy(values, 0, b, 0, values.length);
+```
+%tab-end
+
+%tab "MATLAB"
+```MATLAB
+values = wb_gps_get_values(gps);
+
+% OK, to read the values they should never be explicitly deleted by the controller code
+wb_console_print(sprintf('MY_ROBOT is at position: %g %g %g\n', values(1), values(2), values(3)), WB_STDOUT);
+
+% OK, to copy the values
+x = values[1];
+y = values[2];
+z = values[3];
+
+% OK, another way to copy the values
+a = copy(values);
+```
+%tab-end
+%end
+
+And here are incorrect examples applicable in languages that support constness and memory handling (merely C and C++).
 
 ```c
-const double *pos = wb_gps_get_values(gps);
+const double *values = wb_gps_get_values(gps);
 
-pos[0] = 3.5;      // ERROR: assignment of read-only location
-double a = pos[3]; // ERROR: index out of range
-delete [] pos;     // ERROR: illegal free
-free(pos);         // ERROR: illegal free
+values[0] = 3.5;      // ERROR: assignment of read-only location
+double a = values[3]; // ERROR: index out of range
+delete [] values;     // ERROR: illegal free
+free(values);         // ERROR: illegal free
```

Comment:
I would also remove this, as it is not related with Webots, but more C programming...